packages/patternfly-4/react-core/src/components/ChipGroup/ChipGroup.js

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { css } from '@patternfly/react-styles';
+import PropTypes from 'prop-types';
+import styles from '@patternfly/patternfly/components/ChipGroup/chip-group.css';
+import { Chip } from '@patternfly/react-core';
+
+class ChipGroup extends React.Component {
+  state = {
+    isOpen: false
+  };
+
+  toggleCollapse = () => {
+    this.setState(prevState => ({
+      isOpen: !prevState.isOpen
+    }));
+  };
+
+  renderToolbarGroup() {
+    const { isOpen } = this.state;
+    return <InnerChipGroup {...this.props} isOpen={isOpen} onToggleCollapse={this.toggleCollapse} />;
+  }
+
+  renderChipGroup() {
+    const { className } = this.props;
+    const { isOpen } = this.state;
+    return (
+      <div className={css(styles.chipGroup, className)}>
+        <InnerChipGroup {...this.props} isOpen={isOpen} onToggleCollapse={this.toggleCollapse} />
+      </div>
+    );
+  }
+
+  render() {
+    const { withToolbar, children } = this.props;
+    if (React.Children.count(children)) {
+      return <React.Fragment>{withToolbar ? this.renderToolbarGroup() : this.renderChipGroup()}</React.Fragment>;
+    }
+    return null;
+  }
+}
+
+const InnerChipGroup = props => {
+  const { children, expandedText, isOpen, onToggleCollapse, moreString } = props;
+  const collapsedText = `${React.Children.count(children) - 1} ${moreString}`;
+  return (
+    <React.Fragment>
+      {isOpen ? (
```

Comment:
You already have a Fragment container, so I think this could just be `{isOpen ? children : React.Children.map()`